Hermes/appscale/hermes/stats/stats_app.py

```diff
@@ -1,54 +1,49 @@
-""" Module responsible for configuring Stats API and stats publishing. """
+""" Module responsible for configuring Stats API and stats profiling. """
+import time
+from datetime import datetime
+
 import attr
-from appscale.common import appscale_info
+from tornado.ioloop import PeriodicCallback, IOLoop
 
-from appscale.hermes import Respond404Handler
-from appscale.hermes.stats.constants import (
-  UPDATE_NODE_STATS_INTERVAL, UPDATE_PROCESSES_STATS_INTERVAL,
-  UPDATE_PROXIES_STATS_INTERVAL, UPDATE_CLUSTER_NODES_STATS_INTERVAL,
-  UPDATE_CLUSTER_PROCESSES_STATS_INTERVAL, UPDATE_CLUSTER_PROXIES_STATS_INTERVAL,
-  NODE_STATS_CACHE_SIZE, PROCESSES_STATS_CACHE_SIZE, PROXIES_STATS_CACHE_SIZE,
-  CLUSTER_NODES_STATS_CACHE_SIZE, CLUSTER_PROCESSES_STATS_CACHE_SIZE,
-  CLUSTER_PROXIES_STATS_CACHE_SIZE
-)
+from appscale.hermes.handlers import Respond404Handler
+from appscale.hermes.stats import profile
+from appscale.hermes.stats.constants import PROFILE_NODES_STATS_INTERVAL, \
+  PROFILE_PROCESSES_STATS_INTERVAL, PROFILE_PROXIES_STATS_INTERVAL
 from appscale.hermes.stats.converter import IncludeLists
 from appscale.hermes.stats.handlers import (
-  CachedStatsHandler, CurrentStatsHandler
+  CurrentStatsHandler, CurrentClusterStatsHandler
 )
-from appscale.hermes.stats.handlers import ClusterStatsHandler
 from appscale.hermes.stats.producers.cluster_stats import (
-  ClusterNodesStatsSource, ClusterProxiesStatsSource,
-  ClusterProcessesStatsSource
+  ClusterNodesStatsSource, ClusterProcessesStatsSource,
+  ClusterProxiesStatsSource
 )
 from appscale.hermes.stats.producers.node_stats import NodeStatsSource
 from appscale.hermes.stats.producers.process_stats import ProcessesStatsSource
 from appscale.hermes.stats.producers.proxy_stats import ProxiesStatsSource
-from appscale.hermes.stats.pubsub_base import StatsPublisher
-from appscale.hermes.stats.subscribers.cache import (
-  StatsCache, ClusterStatsCache
-)
-from appscale.hermes.stats.subscribers.profile import (
-  ClusterNodesProfileLog, ClusterProxiesProfileLog, ClusterProcessesProfileLog
-)
 
 
-@attr.s
-class LocalStats(object):
-  """ Container for high level information related to local stats """
-  cache_size = attr.ib()
-  update_interval = attr.ib()
-  cache = attr.ib(default=None)
-  publisher = attr.ib(default=None)
-
-
-@attr.s
-class ClusterStats(object):
-  """ Container for high level information related to cluster stats """
-  cache_size = attr.ib()
-  update_interval = attr.ib()
-  cache = attr.ib(default=None)
-  publisher = attr.ib(default=None)
-  included_field_lists = attr.ib(default=None)
+# This constant can be defined when all stat models and
+# converter.IncludeLists are imported. So it can't be in constants module
+DEFAULT_INCLUDE_LISTS = IncludeLists({
+  # Node stats
+  'node': ['utc_timestamp', 'cpu', 'memory',
```

Comment:
Good point. I'm not sure that it's critical to improve.. Here is my excuse =) :
Include lists explains what fields of stats entities should be included. utc_timestamp is an attribute of node stats entity, but it is not an attribute of process stats entity and proxy stats entity, because for process stats and proxies stats utc_timestamp is shared between number of process stats and proxy stats entities and so this attribute was moved to ProcessesStatsSnapshot and ProxiesStatsSnapshot which have utc_timestamp and list of corresponding stats objects.
When instance of ProcessesStatsSnapshot of ProxiesStatsSnapshot is rendered, utc_timestamp and list of entities are always included so there is no need to include them implicitly.